crates/bevy_ecs/src/entity/mod.rs

```diff
@@ -250,8 +256,8 @@ impl Entities {
         // Use one atomic subtract to grab a range of new IDs. The range might be
         // entirely nonnegative, meaning all IDs come from the freelist, or entirely
         // negative, meaning they are all new IDs to allocate, or a mix of both.
-        let range_end = self.free_cursor.fetch_sub(count as i64, Ordering::Relaxed);
-        let range_start = range_end - count as i64;
+        let range_end = self.free_cursor.fetch_sub(count as Int, Ordering::Relaxed);
```

Comment:
This has saturating behaviour on overflow, which means that we don't fail gracefully when `count` overflows an `i32`.
This one is potentially worse, since it's possible we rely on `ReserveEntitiesIterator` to give exactly the requested number of entities somewhere.
```suggestion
        let range_end = self.free_cursor.fetch_sub(count.try_into().unwrap(), Ordering::Relaxed);
```
which is semantically identical as the current behaviour on normal platforms, but avoids this pitfall. This might also fix the other comment, but I haven't thought this through properly

Reply:
Per https://github.com/bevyengine/bevy/issues/3899#issuecomment-1094361070, I think we should drop comments above this unwrap explaining why this could panic.